torch/_subclasses/functional_tensor.py

```diff
@@ -401,7 +409,7 @@ def unset_functional_temporarily():
 # - Doing so means that it does not automatically compose with other
 #   functorch transforms, since these transforms always run above __torch_dispatch__.
 #   That's why this util lives here, and not in functorch.
-def dispatch_functionalize(func):
+def dispatch_functionalize(func, mode=None):
```

Comment:
We should ban passing in mode as follow up. Didn't want to mess with auto_functionalize in this PR

Reply:
tbh I don't think it would be a much larger PR if you use `_detect_functional_mode()` necessary directly in this PR (and then you don't have to worry about the fact that you're changing the API for `dispatch_functionalize` only to have to unwind it later)